### PR TITLE
Expose the left-handed matrix creation APIs

### LINKED ISSUE #80332: Provide "Left Handed" overloads of matrix create functions
## Background and Motivation

Matrix4x4 currently exposes several `Create` functions that operate exclusively in the "right-handed" domain. In practice, various graphics opt to use a "left-handed" domain instead and similary numerics libraries (such as DirectX Math or GLM) typically provide overloads for both domains. 

Additionally, we have had several issues where users have been confused about which domain the `Matrix4x4` functions default to and have had prior requests for providing such functionality.

## Proposed API

```csharp
namespace System.Numerics;

public partial struct Matrix4x4
{
    public static Matrix4x4 CreateLookAtLH(Vector3 cameraPosition, Vector3 cameraTarget, Vector3 cameraUpVector);

    public static Matrix4x4 CreateLookTo(Vector3 cameraPosition, Vector3 cameraDirection, Vector3 cameraUpVector);
    public static Matrix4x4 CreateLookToLH(Vector3 cameraPosition, Vector3 cameraDirection, Vector3 cameraUpVector);

    public static Matrix4x4 CreateOrthographicLH(float width, float height, float zNearPlane, float zFarPlane);
    public static Matrix4x4 CreateOrthographicOffCenterLH(float left, float right, float bottom, float top, float zNearPlane, float zFarPlane);

    public static Matrix4x4 CreatePerspectiveLH(float width, float height, float nearPlaneDistance, float farPlaneDistance);
    public static Matrix4x4 CreatePerspectiveFieldOfViewLH(float fieldOfView, float aspectRatio, float nearPlaneDistance, float farPlaneDistance);
    public static Matrix4x4 CreatePerspectiveOffCenterLH(float left, float right, float bottom, float top, float nearPlaneDistance, float farPlaneDistance);
    
    public static Matrix4x4 CreateViewportLH(float x, float y, float width, float height, float minDepth, float maxDepth)
}
```

## Additional Considerations

The documentation should be clearly updated to indicate the non-suffixed functions are "right-handed" and the new functions are "left-handed".

It may be desirable to spell out "LeftHanded", however the common case is abbreviating and within the context of Matrix4x4 it should be self-explanatory.

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #80332

### CODE REVIEW

File: src/libraries/System.Numerics.Vectors/tests/Matrix4x4Tests.cs [hunk 1]
```diff
@@ -868,30 +867,142 @@ public void Matrix4x4CreateLookAtTest()
             Vector3 cameraUpVector = new Vector3(0.0f, 1.0f, 0.0f);
 
             Matrix4x4 expected = new Matrix4x4();
-            expected.M11 = 0.979457f;
-            expected.M12 = -0.0928267762f;
-            expected.M13 = 0.179017f;
```
Comment (tannergooding) [inline src/libraries/System.Numerics.Vectors/tests/Matrix4x4Tests.cs:873]: There's no change in these values, they've just been simplified to the shortest roundtrippable string for consistency

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs [hunk 2]
```diff
@@ -448,24 +547,42 @@ public static Matrix4x4 CreateTranslation(Vector3 position)
...
         /// <param name="minDepth">Viewport minimum depth.</param>
         /// <param name="maxDepth">Viewport maximum depth.</param>
-        /// <returns>The viewport matrix.</returns>
+        /// <returns>The right-handed viewport matrix.</returns>
         /// <remarks>
         /// Viewport matrix
         /// |   width / 2   |        0       |          0          | 0 |
         /// |       0       |   -height / 2  |          0          | 0 |
-        /// |       0       |        0       | maxDepth - minDepth | 0 |
+        /// |       0       |        0       | minDepth - maxDepth | 0 |
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs:562]: This is a minor bug fix to the new API. It was creating a left-handed viewport.

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs [hunk 3]
```diff
@@ -451,9 +452,47 @@ public static Impl CreateLookAt(in Vector3 cameraPosition, in Vector3 cameraTarg
...
+                    Vector3.Dot(axisY, negativeCameraPosition),
+                    Vector3.Dot(axisZ, negativeCameraPosition),
+                    1
+                );
+
+                return result;
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Impl CreateLookToLeftHanded(in Vector3 cameraPosition, in Vector3 cameraDirection, in Vector3 cameraUpVector)
```
Comment (krwq) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs:465]: why can't this just pass negative cameraDirection to the one above?

Comment (krwq) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs:465]: (similar question to other methods)

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs:465]: This one in particular could, but that adds a hidden additional cost that the JIT can't currently optimize away. For the others the algorithms are not simply negating a single input.

Comment (krwq) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs:465]: is there an issue against JIT to improve this? is this cost high enough to justify code duplication?

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs:465]: > is there an issue against JIT to improve this

There are many issues tracking the broader issues that impact this space. It's not really limited to just one thing.

> is this cost high enough to justify code duplication?

This is a known perf oriented type and the amount of code duplication is not significant. We have (ignoring the formatting done to help improve readability, around 8-11 lines of pretty simple code here.

**Review by michaelgsharp — APPROVED** [review]

**Pull request merged**